strict_encoding/src/monero.rs

```diff
@@ -0,0 +1,258 @@
+}
+
+impl Strategy for RctType {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for PublicKey {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for PrivateKey {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Hash {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Hash8 {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Address {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for BoroSig {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Bulletproof {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Clsag {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for CtKey {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Key64 {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Key {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for MgSig {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for MultisigKlrki {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for MultisigOut {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for RangeSig {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for RctSigBase {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for Signature {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl Strategy for VarInt {
+    type Strategy = strategies::MoneroConsensus;
+}
+
+impl StrictEncode for Amount {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        self.as_pico().strict_encode(&mut e)
+    }
+}
+
+impl StrictDecode for Amount {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        Ok(Self::from_pico(u64::strict_decode(&mut d)?))
+    }
+}
+
+impl StrictEncode for SignedAmount {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        self.as_pico().strict_encode(&mut e)
+    }
+}
+
+impl StrictDecode for SignedAmount {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        Ok(Self::from_pico(i64::strict_decode(&mut d)?))
+    }
+}
+
+impl StrictEncode for PaymentId {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        Ok(e.write(&self.to_fixed_bytes())?)
+    }
+}
+
+impl StrictDecode for PaymentId {
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+        let mut bytes = [0u8; 8];
+        d.read_exact(&mut bytes)?;
+        Ok(Self::from(bytes))
+    }
+}
+
+impl StrictEncode for Index {
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Error> {
+        self.major.strict_encode(&mut e)?;
+        self.minor.strict_encode(e)
```

Comment:
There is a macro for that `strict_encode_list!(e; self.major, self.minor)`

Reply:
Done in f7d17021bd44ec9630adbca916bf341865866079